### PR TITLE
Add regression test for missing_constant with whitespace suppressing min/max range errors

### LINKED ISSUE #1656: validate WARNs if value is below valid_minimum but equals missing_constant
### Checked for duplicates

No - I haven't checked

### 🐛 Describe the bug

validate -t attachedfile.xml gives 25 WARNINGs that the fields have values that are below the minimum value, e.g. the last field in the .CSV file is -1.000, the .xml has
`          <Special_Constants>
            <missing_constant>   -1.000</missing_constant>
            <valid_maximum>   10000.000</valid_maximum>
            <valid_minimum>       0.001</valid_minimum>
          </Special_Constants>
`
and validate for that field says 
`      WARNING  [warning.table.field_value_out_of_special_constant_min_max_range]   data object 1, record 1, field 74: Field has a value '-1' that is less than the defined minimum value '0.001'. 
`

Is this a bug? We've had issues about this kind of typecasting, but I can't tell what the answer is. If it's not a bug, i.e. if validate is behaving as it should, how should the user specify missing_constant? Either of the following works, but I should probably explain why
            <missing_constant>   -1</missing_constant>
            <missing_constant>-1</missing_constant>

### 🕵️ Expected behavior

I expected no validate warnings or errors

### 📜 To Reproduce

1.
2.
3.
...

### 🖥 Environment Info

- Version of this software [e.g. vX.Y.Z]
- Operating System: [e.g. MacOSX with Docker Desktop vX.Y]
...

### 📚 Version of Software Used

   Version      4.2.0
   Date         2026-08-17T18:54:02Z

### 🩺 Test Data / Additional context

[Archive.zip](https://github.com/user-attachments/files/31151503/Archive.zip)

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### 🎉 Integration & Test

_No response_

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Adds a regression test for #1656, which reports that `validate` emits false warnings/errors when a `missing_constant` value contains leading or trailing whitespace (e.g., `   -1.000`) and the actual field value is the numeric equivalent (`-1.000`). The fix for #1660 likely resolves this as well — this test case ensures #1656 stays covered going forward.

**Test data** (`src/test/resources/github1656/`):
- `test_missing_constant_whitespace.csv` — 3-record delimited table; two records contain the missing-constant sentinel value `-1.000`, one contains a valid value `5.000`
- `test_missing_constant_whitespace.xml` — label with `<missing_constant>   -1.000</missing_constant>` (leading whitespace) and `Field_Statistics.minimum = 0.001`, which triggers false `field_value_out_of_min_max_range` errors on unpatched code

**Confirmed bug reproduction** against v4.2.0-SNAPSHOT: 2 `error.table.field_value_out_of_min_max_range` errors are emitted for rows containing the missing-constant value.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 90%

## ⚙️ Test Data and/or Report

New Cucumber scenario added to `src/test/resources/features/4.2.x.feature` (test 1656-1):

```
--skip-context-validation -t {datasrc}/test_missing_constant_whitespace.xml
```

Expected outcome after fix: `passed=1, totalErrors=0, totalWarnings=2` (only the two expected schema-related warnings matching the same pattern as test 1660-1).

Current outcome on v4.2.0-SNAPSHOT (bug reproduced):
```
2  error.table.field_value_out_of_min_max_range
1  warning.label.bad_schematypens
1  warning.label.missing_schematron_spec
```

Run with:
```bash
mvn test -Dtest=\!ReferenceIntegrityTest* -Dcucumber.filter.tags='@4.2.x'
```

## ♻️ Related Issues

Fixes #1656
Refs #1660

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: src/main/java/gov/nasa/pds/tools/validate/SpecialConstantChecker.java [hunk 1]
```diff
@@ -28,19 +28,22 @@ public static boolean isInfOrNan (String value) {
...
-      matched |= value.equals(constants.getNotApplicableConstant());
-      matched |= value.equals(constants.getSaturatedConstant());
-      matched |= value.equals(constants.getUnknownConstant());
+      matched |= equalsStripped(value, constants.getErrorConstant());
+      matched |= equalsStripped(value, constants.getInvalidConstant());
+      matched |= equalsStripped(value, constants.getMissingConstant());
+      //matched |= equalsStripped(value, constants.getHighInstrumentSaturation());
+      //matched |= equalsStripped(value, constants.getHighRepresentationSaturation());
+      //matched |= equalsStripped(value, constants.getLowInstrumentSaturation());
+      //matched |= equalsStripped(value, constants.getLowRepresentationSaturation());
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/tools/validate/SpecialConstantChecker.java:43]: Choose your own adventure! You encounter some commented-out code. Do you:

1. Add a comment above the code explaining _why_ these 4 lines are commented out; or …
2. Delete these 4 lines?

Both paths may lead to success … what do you decide? 🧙

**Review by nutjob4life — CHANGES_REQUESTED** [review]
The `strip` seems to do the trick! And the tests agree; they all pass.

However, there's a block of mysterious, commented-out code (see below).

File: src/main/java/gov/nasa/pds/tools/validate/SpecialConstantChecker.java (hunk 1 above)
Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/tools/validate/SpecialConstantChecker.java]: @nutjob4life will clean up. Looks like this was leftover undocumented comments

**Review by nutjob4life — APPROVED** [review]
Deltas: ✓
Tests: ✓
Approval: ✅
Test details:
```
[INFO] Tests run: 329, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:23 min
```
